[vault] Version bump to 0.8.3, add health check hook and templatize the r…

predominant opened the pull request:
…un hook

Signed-off-by: Graham Weldon <graham@grahamweldon.com>

thesentinels commented:
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

predominant commented:
This is ready for review now. pre-commit checks were complaining about newlines, and some spaces in openssl plan, which is now fixed.

johnjelinek commented on vault/hooks/health_check:
```diff
@@ -0,0 +1,20 @@
+
+protocol=http{{#unless cfg.listener.tls_disable}}s{{/unless}}
+
+status=$({{pkgPathFor "core/curl"}}/bin/curl \
```
since `core/curl` is a runtime dep, it should be in your `PATH` so you shouldn't have to `pkgPathFor` it at the time your `health_check` runs.

predominant replied:
You're absolutely right. I've pushed a commit to change it.

predominant commented:
I've fixed the merge conflict, and rewrote that unverified commit. 👍

predominant commented:
@eeyun Not only have i removed that (silly) OpenSSL file change, i've rebased against master, and squashed all the commits. Hows that for service!

![tenor-84065716](https://user-images.githubusercontent.com/24568/32638900-afad0a9e-c604-11e7-842e-d0a5c9ff6070.gif)
